modules/messages.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Manages backup, clean and restore of device SMS and MMS messages.
+ */
+var Messages = function() {};
+
+/**
+ * @access public
+ * @description Backups the current SMS and MMS messages on the device
+ *   to external storage. Callback is invoked upon completion. If an error
+ *   occurred, it will be passed as the first parameter to the callback.
+ * @param {callback} callback
+ */
+Messages.prototype.backup = function(callback) {
+  var that = this;
+
+  this._getMessages(function(msgs) {
+    var msgsJSON = [];
+
+    if (msgs.error) {
+      callback(msgs.error);
+      return;
+    }
+
+    for (var i = 0; i < msgs.list.length; i++) {
+      var msg;
+
+      if (msgs.list[i] instanceof MozSmsMessage) {
```

Comment:
Seems like a lot of the SMS and MMS functionality could be factored out so that the if and else branches have only the fields specific to their type?